OPAL/br/src/main/scala/org/opalj/br/fpcf/analyses/immutability/ClassImmutabilityAnalysis.scala

```diff
@@ -350,33 +318,30 @@ class ClassImmutabilityAnalysis(val project: SomeProject) extends FPCFAnalysis {
 
                 // Field Immutability related dependencies:
                 case FinalP(TransitivelyImmutableField) =>
+
                 case FinalP(NonTransitivelyImmutableField) =>
                     maxLocalImmutability = NonTransitivelyImmutableClass
-                case FinalP(MutableField) => return Result(t, MutableClass);
-                case UBP(MutableField)    => return Result(t, MutableClass);
-                case ELBP(e, NonTransitivelyImmutableField |
-                    TransitivelyImmutableField) => dependees -= e
-                case UBP(TransitivelyImmutableField)    => // no information about field mutability
-                case UBP(NonTransitivelyImmutableField) => maxLocalImmutability = NonTransitivelyImmutableClass
+
+                case FinalP(MutableField)                                            => return Result(t, MutableClass);
+                case UBP(MutableField)                                               => return Result(t, MutableClass);
+
+                case LBP(NonTransitivelyImmutableField | TransitivelyImmutableField) =>
+
+                case UBP(TransitivelyImmutableField)                                 => // no information about field mutability
+
+                case UBP(NonTransitivelyImmutableField)                              => maxLocalImmutability = NonTransitivelyImmutableClass
+
                 case UBP(DependentlyImmutableField(parameter)) if maxLocalImmutability != NonTransitivelyImmutableClass =>
                     genericTypeParameters ++= parameter
-                    maxLocalImmutability =
-                        DependentlyImmutableClass(genericTypeParameters)
-                case _ => Result(t, MutableClass) // TODO
+                    maxLocalImmutability = DependentlyImmutableClass(genericTypeParameters)
+
+                case _ => Result(t, MutableClass) //TODO
```

Comment:
What is "TODO" here?

Reply:
I wondered as well...

Reply:
I looked into its history. It once was "TODO check", which is only that tad more meaningful. One of your colleagues removed and readded this TODO (the latter in a reversion commit).

Given it is very old and not meaningful, i think we can remove it.